dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -2,203 +2,277 @@
+    static final String INCLUDE = "include";
+    static final String PATHS = "paths";
+
+    private final ESIndexAPI esIndexAPI;
+    private final IndiciesAPI indicesAPI;
+    private final SiteSearchAPI siteSearchAPI;
+    private final HostAPI hostAPI;
+    private final UserAPI userAPI;
+    private final SiteSearchAuditAPI siteSearchAuditAPI;
+    private final PublisherAPI publisherAPI;
+
+    private String bundleId;
+
+    @VisibleForTesting
+    SiteSearchJobImpl(
+            final ESIndexAPI esIndexAPI,
+            final IndiciesAPI indicesAPI,
+            final SiteSearchAPI siteSearchAPI,
+            final HostAPI hostAPI,
+            final UserAPI userAPI,
+            final SiteSearchAuditAPI siteSearchAuditAPI,
+            final PublisherAPI publisherAPI
+            ) {
+        this.esIndexAPI = esIndexAPI;
+        this.indicesAPI = indicesAPI;
+        this.siteSearchAPI = siteSearchAPI;
+        this.hostAPI = hostAPI;
+        this.userAPI = userAPI;
+        this.siteSearchAuditAPI = siteSearchAuditAPI;
+        this.publisherAPI = publisherAPI;
+    }
+
+    public SiteSearchJobImpl() {
+        this(APILocator.getESIndexAPI(), APILocator.getIndiciesAPI(), APILocator.getSiteSearchAPI(),
+                APILocator.getHostAPI(), APILocator.getUserAPI(),
+                APILocator.getSiteSearchAuditAPI(), APILocator.getPublisherAPI());
+    }
+
     private PublishStatus status = new PublishStatus();
     public PublishStatus getStatus() {
         return status;
     }
     public void setStatus(PublishStatus status) {
         this.status = status;
     }
+
+    @VisibleForTesting
+    public String getBundleId() {
+        return bundleId;
+    }
+
     @SuppressWarnings("unchecked")
-    public void run(JobExecutionContext jobContext) throws JobExecutionException, DotPublishingException, DotDataException, DotSecurityException, ElasticsearchException, IOException {
-        if(LicenseUtil.getLevel() < LicenseLevel.STANDARD.level)
+    public void run(final JobExecutionContext jobContext) throws JobExecutionException, DotPublishingException, DotDataException, DotSecurityException, ElasticsearchException, IOException {
+        if (LicenseUtil.getLevel() < LicenseLevel.STANDARD.level) {
+            Logger.warn(this, "Invalid attempt to run SiteSearch job without a license.");
             return;
-        
+        }
         String date = DateUtil.getCurrentDate();
-        ActivityLogger.logInfo(getClass(), "Job Started", "User:" + APILocator.getUserAPI().getSystemUser().getUserId() + "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
-        AdminLogger.log(getClass(), "Job Started", "User:" + APILocator.getUserAPI().getSystemUser().getUserId() + "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
+        ActivityLogger.logInfo(getClass(), "Job Started",
+                "User:" + userAPI.getSystemUser().getUserId() + "; Date: " + date
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "; Date: " appears 4 times in this file; the first occurrence is on line 127](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093862)